presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -54,15 +54,18 @@ void onFinalFailure(
 
 std::string bodyAsString(
     http::HttpResponse& response,
-    memory::MemoryPool* pool) {
+    memory::MemoryPool* pool,
```

Comment:
If pool is null, then no free? So we don't need immediateBufferTransfer?

Reply:
this pool is passed in from PrestoExchangeSource which will always be non-null. So we need to rely on the immediateBufferTransfer flag to tell if we shall free. Note that only inside HttpClient/HttpResponse we can tell based on nullness of pool

Reply:
Can callsite of bodyAsString to determine pass pool or not based on immediateBufferTransfer? So the callee doesn't need to know the details and simply based on if pool is null or not? Thanks!